File: U4/C#/U4P_10.cs
using System;

namespace MyApp // Note: actual namespace depends on the project name.
{
    internal class U4P_10
    {
        static void Main(string[] args)
        {
            string[] paises = { "Argentina", "Brasil", "Chile", "México", "Perú" };
            double longitudPromedio = CalcularLongitudPromedio(paises);
            Console.WriteLine("Arreglo de nombres de países:");
            ImprimirArreglo(paises);
            Console.WriteLine("\nLongitud promedio de las palabras: " + longitudPromedio);
        }

        static double CalcularLongitudPromedio(string[] arreglo)
        {
            int totalCaracteres = 0;
            foreach (string palabra in arreglo)
            {
                totalCaracteres += palabra.Length;
            }
            return (double)totalCaracteres / arreglo.Length;
        }

        static void ImprimirArreglo(string[] arreglo)
        {
            foreach (string elemento in arreglo)
            {
                Console.Write(elemento + " ");
            }
            Console.WriteLine();
        }
    }
}

File: U4/Java/U4P_10.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 */

package com.mycompan.u4p_10;

/**
 *
 * @author alfre
 */
public class U4P_10 {

    public static void main(String[] args) {
        String[] paises = { "Argentina", "Brasil", "Chile", "México", "Perú" };
        double longitudPromedio = calcularLongitudPromedio(paises);
        System.out.println("Arreglo de nombres de países:");
        imprimirArreglo(paises);
        System.out.println("\nLongitud promedio de las palabras: " + longitudPromedio);
    }

    static double calcularLongitudPromedio(String[] arreglo) {
        int totalCaracteres = 0;
        for (String palabra : arreglo) {
            totalCaracteres += palabra.length();
        }
        return (double) totalCaracteres / arreglo.length;
    }

    static void imprimirArreglo(String[] arreglo) {
        for (String elemento : arreglo) {
            System.out.print(elemento + " ");
        }
        System.out.println();
    }
}